starter/src/main/java/com/starter/fullstack/rest/InventoryController.java

```diff
@@ -38,17 +39,27 @@ public List<Inventory> findInventories() {
     return this.inventoryDAO.findAll();
   }
 
+  @GetMapping("/getById")
+  public Optional<Inventory> findInventoryById(String id) {
+    return  this.inventoryDAO.retrieve(id);
+  }
+
   @PostMapping
   public Inventory saveInventory(@Valid @RequestBody Inventory inventory) {
     return this.inventoryDAO.create(inventory);
   }
 
+  @PostMapping("/update")
+  public Optional<Inventory> updateInventory(@Valid @RequestBody String id, @Valid @RequestBody Inventory inventory) {
```

Comment:
You usually don't want multiple request bodies for an endpoint. Change id to be a @RequestParam. You will need to slightly change the front end for it to work.